bind PUT causes broker panic and 502 when ServiceInstance not found

**What happened**:
I tried to ```PUT /v2/service_instances/$INSTANCE_ID/service_bindings/$BINDING_ID?accepts_incomplete=true``` where ```$INSTANCE_ID``` does not exist.

The broker panics and returns 502.

```
[2018-01-17T21:47:16.399Z] [DEBUG] - Unable to find originating origin header
[2018-01-17T21:47:16.4Z] [ERROR] - Could not find a service instance in dao - 100: Key not found (/service_instance/43e76ef3-7e2a-43ee-b1fe-845a315bf557) [285]
[2018-01-17T21:47:16.4Z] [DEBUG] - Auto Escalate has been set to true, we are escalating permissions
[2018-01-17T21:47:16.4Z] [DEBUG] - Injecting PlanID as parameter: { _apb_plan_id: dev }
[2018-01-17T21:47:16.4Z] [DEBUG] - Injecting ServiceClassID as parameter: { _apb_service_class_id: dh-postgresql-apb-s964m }
[2018-01-17T21:47:16.4Z] [DEBUG] - Injecting ServiceInstanceID as parameter: { _apb_service_instance_id:  }
[2018-01-17T21:47:16.402Z] [INFO] - ASYNC binding in progress
[2018-01-17T21:47:16.402Z] [DEBUG] - bindjob: binding job started, calling apb.Bind
[2018-01-17T21:47:16.402Z] [NOTICE] - ============================================================
[2018-01-17T21:47:16.402Z] [NOTICE] -                        BINDING    
[2018-01-17T21:47:16.402Z] [NOTICE] - ============================================================
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1511412]
goroutine 326 [running]:
github.com/openshift/ansible-service-broker/pkg/apb.Bind(0xc4206dc840, 0xc420168108, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/mhrivnak/golang/src/github.com/openshift/ansible-service-broker/pkg/apb/bind.go:32 +0x152
github.com/openshift/ansible-service-broker/pkg/broker.(*BindingJob).Run(0xc420732270, 0xc4207322a0, 0x24, 0xc420244900)
    /home/mhrivnak/golang/src/github.com/openshift/ansible-service-broker/pkg/broker/binding_job.go:53 +0x94
created by github.com/openshift/ansible-service-broker/pkg/broker.(*WorkEngine).StartNewJob
    /home/mhrivnak/golang/src/github.com/openshift/ansible-service-broker/pkg/broker/work_engine.go:64 +0x115
```

**What you expected to happen**:
400 per the OSB API spec

**How to reproduce it**:
Configure the broker to run APBs for bind and unbind, and to auto-escalate, then run the below script.

There is likely a similar failure even when APBs are not run for bind.

```
#!/usr/bin/env bash

INSTANCE_ID=$(uuidgen)
BINDING_ID=$(uuidgen)
PLAN_UUID="7f4a5e35e4af2beb70076e72fab0b7ff"
SERVICE_UUID="dh-postgresql-apb-s964m"

req="{
  \"plan_id\": \"$PLAN_UUID\",
  \"service_id\": \"$SERVICE_UUID\",
  \"context\": \"blog-project\",
  \"app_guid\":\"\",
  \"bind_resource\":{},
  \"parameters\": {}
}"

curl \
    -k \
    -i \
    -X PUT \
    -H "Authorization: bearer $(oc whoami -t)" \
    -H "Content-type: application/json" \
    -H "Accept: application/json" \
    -H "X-Broker-API-Originating-Identity: " \
    -d "$req" \
    "https://asb-1338-ansible-service-broker.172.17.0.1.nip.io/ansible-service-broker/v2/service_instances/$INSTANCE_ID/service_bindings/$BINDING_ID?accepts_incomplete=true"
```

--- a/pkg/handler/handler.go
+++ b/pkg/handler/handler.go
@@ -537,6 +537,7 @@ func (h handler) bind(w http.ResponseWriter, r *http.Request, params map[string]
 		default:
 			writeResponse(w, http.StatusInternalServerError, broker.ErrorResponse{Description: err.Error()})
 		}
+		return
 	}
 
 	if !h.brokerConfig.GetBool("broker.auto_escalate") {

--- a/pkg/handler/handler_test.go
+++ b/pkg/handler/handler_test.go
@@ -35,10 +35,11 @@ import (
 )
 
 type MockBroker struct {
-	Name      string
-	Verify    map[string]bool
-	Err       error
-	Operation string
+	Name                  string
+	Verify                map[string]bool
+	Err                   error
+	Operation             string
+	getServiceInstanceErr error
 }
 
 type MockUserServiceAdapter struct {
@@ -106,7 +107,7 @@ func (m MockBroker) LastOperation(uuid.UUID, *broker.LastOperationRequest) (*bro
 }
 
 func (m MockBroker) GetServiceInstance(uuid.UUID) (apb.ServiceInstance, error) {
-	return apb.ServiceInstance{}, nil
+	return apb.ServiceInstance{}, m.getServiceInstanceErr
 }
 
 func (m MockBroker) GetBindInstance(uuid.UUID) (apb.BindInstance, error) {
@@ -365,7 +366,8 @@ func TestBindAlreadyProvisioned(t *testing.T) {
 }
 
 func TestBindNotFound(t *testing.T) {
-	testhandler, w, r, params := buildBindHandler(uuid.New(), broker.ErrorNotFound)
+	testhandler, w, r, params := buildBindHandler(uuid.New(), nil)
+	testhandler.broker = MockBroker{Name: "testbroker", Err: nil, getServiceInstanceErr: broker.ErrorNotFound}
 	testhandler.bind(w, r, params)
 	ft.AssertEqual(t, w.Code, 400, "should've been a bad request for error not found")
 	ft.AssertError(t, w.Body, "not found")

